
int8 axis
float64 desiredRate